polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisSecretsManager.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.auth;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.polaris.core.PolarisCallContext;
+import org.apache.polaris.core.entity.PolarisPrincipalSecrets;
+import org.apache.polaris.core.persistence.BaseResult;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Manages secrets for Polaris principals. */
+public interface PolarisSecretsManager {
```

Comment:
Once we have other types of stored secrets for outbound integrations/connections, we might have a different interface like "PolarisIntegrationSecretsManager" -- are you envisioning combining all different types of secrets into this interface, or would we split first-party principal auth-related management here (e.g. we might call this one `PolarisPrincipalSecretsManager` and keep it under this `auth` directory) from outbound integration secret management, possibly keeping `PolarisIntegrationSecretsManager` under the `persistence` directory?